src/rust/engine/process_execution/src/lib.rs

```diff
@@ -444,15 +444,16 @@ impl ProcessExecutionStrategy {
             Self::RemoteExecution(_) => "remote_execution".to_string(),
             // NB: this image will include the container ID, thanks to
             // https://github.com/pantsbuild/pants/pull/17101.
-            Self::Docker(image) => format!("docker_execution: {image}"),
+            // NB: We don't include the mount info in the cache.
+            Self::Docker(image, _) => format!("docker_execution: {image}"),
```

Comment:
This is the only thing I'm not 100% on. Any thoughts?

Reply:
From my local hacking, I think maybe including it in they key would be good for edits.

Reply:
OK, so I [changed this](https://github.com/pantsbuild/pants/pull/20322/commits/4d8df685eb006f7622c85a906ae5bd76fff094c7) to use the bind mounts here, but I'm not seeing Pants re-start the container

Reply:
[Ah solved it](https://github.com/pantsbuild/pants/pull/20322/commits/d3279b73c0a4f48629477e5fca059e0d28d7439e)

Keeping this open, since the original comment now applies again :smile:

Reply:
Omitting info from the process cache keys seems a bit dangerous, and something that has to be done carefully. (I'm not the right person to make a final call about the consequences.)